src/main/scala/com/redislabs/provider/redis/redisFunctions.scala

```diff
@@ -390,6 +402,28 @@ object RedisContext extends Serializable {
     conn.close()
   }
 
+  /**
+    * @param arr hashName/k/vs which should be saved in the target host
+    *            save all the k/vs to hashName(list type) to the target host
+    * @param ttl time to live
+    */
+  def setHash2(arr: Iterator[(String, (String, String))], ttl: Int, redisConfig: RedisConfig,
+              readWriteConfig: ReadWriteConfig) {
+    implicit val rwConf: ReadWriteConfig = readWriteConfig
+    arr.map(h => (h._1, h)).toArray.groupBy(_._1).
+      mapValues(a => a.map(p => p._2)).foreach { x =>
```

Comment:
@charsyam Pretty cool that you implemented this group by logic so that this function still retrieves 1 connection per unique hash as it is the case in the original `setHash` function 🚀